src/main/java/com/fasterxml/jackson/perf/json/JsonArbitraryFieldNameBenchmark.java

```diff
@@ -0,0 +1,151 @@
+package com.fasterxml.jackson.perf.json;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+public class JsonArbitraryFieldNameBenchmark {
+
+    public enum FactoryMode {
+        DEFAULT() {
+            @Override
+            JsonFactory apply(JsonFactory factory) {
+                return factory;
+            }
+        },
+        NO_INTERN() {
+            @Override
+            JsonFactory apply(JsonFactory factory) {
+                return factory.disable(JsonFactory.Feature.INTERN_FIELD_NAMES);
+            }
+        },
+        NO_CANONICALIZE() {
+            @Override
+            JsonFactory apply(JsonFactory factory) {
+                return factory.disable(JsonFactory.Feature.CANONICALIZE_FIELD_NAMES);
+            }
+        };
+
+        abstract JsonFactory apply(JsonFactory factory);
+    }
+
+    /**
+     * Ideally we would not generate inputs within the measured component of the benchmark.
+     */
+    public enum InputType {
+        INPUT_STREAM() {
+            @Override
+            JsonParser create(JsonFactory factory, Supplier<String> jsonSupplier) throws IOException {
+                return factory.createParser(new ByteArrayInputStream(jsonSupplier.get().getBytes(StandardCharsets.UTF_8)));
```

Comment:
Why not make supplier provide bytes, to avoid additional conversion overhead here?

Reply:
Dynamic part may be problematic, but existing tests have helper class (`InputData`?) that creates data, extracts bytes etc. So it's more a question of not doing this conversion for each parse/generation run than performance of getting bytes itself. That is, keeping it out of measurement loop. If that makes sense.